linux/shim/io.go

```diff
@@ -15,8 +15,9 @@ import (
 )
 
 func copyConsole(ctx context.Context, console console.Console, stdin, stdout, stderr string, wg, cwg *sync.WaitGroup) error {
+	fctx := context.Background()
 	if stdin != "" {
-		in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
+		in, err := fifo.OpenFifo(fctx, stdin, syscall.O_RDONLY, 0)
```

Comment:
When the `RPC` finishes the `ctx.Err()` will always be `context.Canceled`, won't it? In which case the situation would be the same.

In `containerd` case, we only want to honor the cancel state if the `context` is cancelled before we return from the `RPC` handler.